### PR TITLE
[IOTDB-4743]  OperatePipeProcedure support rollback

### PR DESCRIPTION
**Pull request by Cpaulyz**
## Description
* Rename internal PipeStatus to make it easy to understand.
  * `PREPARE_CREATE` -> `PARTIAL_CREATE` （not visible to the user
  * `PREPARE_START` -> `PARTIAL_RUNNING` （not visible to the user
  * `PREPARE_STOP` -> `PARTIAL_STOP` （not visible to the user
  * `PREPARE_DROP` -> `DROP` （visible to the user only if failed to execute drop on partial datanodes
* Implement rollback logic for OperatePipeProcedure
  * Design: https://apache-iotdb.feishu.cn/docx/doxcnpfFAOfJJGj6qmDM4VPEDNf#doxcng82QQoeC2IWMi8CUudmA0d
  * Test: https://apache-iotdb.feishu.cn/docx/NsLPdhkC1oa1QCxKOMlcj9RxntK#Z8AUdYsM0oAesKxy6Pzc3ny7nEg
* Use type instead of .ordinal() in PipeStatus
* Set default SyncDelOp = false

### CODE REVIEW

File: confignode/src/main/java/org/apache/iotdb/confignode/manager/SyncManager.java [hunk 1]
```diff
@@ -180,41 +187,91 @@ public TGetAllPipeInfoResp getAllPipeInfo() {
...
+  /**
+   * Broadcast DataNodes to operate PIPE operation for roll back procedure.
+   *
+   * @param pipeName name of PIPE
+   * @param operation only support {@link SyncOperation#START_PIPE}, {@link SyncOperation#STOP_PIPE}
+   *     and {@link SyncOperation#DROP_PIPE}
+   * @param dataNodeIds target DataNodeId set
+   */
+  public void operatePipeOnDataNodesForRollback(
+      String pipeName, long createTime, SyncOperation operation, Set<Integer> dataNodeIds) {
```
Comment (HTHou) [inline confignode/src/main/java/org/apache/iotdb/confignode/manager/SyncManager.java:221]: Add `@param createTime` in javadoc

Comment (Cpaulyz) [reply confignode/src/main/java/org/apache/iotdb/confignode/manager/SyncManager.java]: done

File: node-commons/src/main/java/org/apache/iotdb/commons/sync/pipe/PipeStatus.java [hunk 2]
```diff
@@ -23,8 +23,8 @@ public enum PipeStatus {
...
   STOP,
   // internal status
-  PREPARE_CREATE,
-  PREPARE_START,
-  PREPARE_STOP,
-  PREPARE_DROP
+  PARTIAL_CREATE,
+  PARTIAL_START,
+  PARTIAL_STOP,
+  PARTIAL_DROP
```
Comment (HTHou) [inline node-commons/src/main/java/org/apache/iotdb/commons/sync/pipe/PipeStatus.java:29]: Why do u change these name?

**Review by neuyilan — APPROVED** [review]
LGTM

**Review by yschengzi — APPROVED** [review]
LGTM

**Pull request merged**